core/src/wallet/cosmos/CosmosLikeAccount.cpp

```diff
@@ -318,13 +320,20 @@ void CosmosLikeAccount::inflateOperation(
     out.currencyName = getWallet()->getCurrency().name;
     out.date = tx.timestamp;
     out.trust = std::make_shared<TrustIndicator>();
+    // Fees are added only on the MSGFEES Message Type.
     auto fees = 0;
     if (cosmos::stringToMsgType(msg.type.c_str()) == api::CosmosLikeMsgType::MSGFEES) {
-        std::for_each(tx.fee.amount.begin(), tx.fee.amount.end(), [&](cosmos::Coin amount) {
+        std::for_each(tx.fee.amount.begin(), tx.fee.amount.end(), [&](const cosmos::Coin &amount) {
             assert(amount.denom == "uatom");  // FIXME Temporary until all units correctly supported
             fees += BigInt::fromDecimal(amount.amount).toInt();
         });
     }
+    // The value is updated to the fees actually paid if the transaction has a GasUsed value to use.
+    if (fees != 0 && tx.gasUsed) {
+        auto feeConsumptionRatio = static_cast<float>(tx.gasUsed.getValue().toInt()) /
+                                   static_cast<float>(tx.fee.gas.toInt());
```

Comment:
Potential division by zero detected here ! 
Can `tx.fee.gas` be zero ? Even if it's not the case now, this invariant can be change in the future and I suggest to protect the code from that UB.

Reply:
Theoretically it cannot be. A 0 gas transaction is either :
- Never broadcast by the node (because of spam protection mechanisms)
- Invalid, as the signature verification (or the `gas per byte` cost) will immediately trigger an `OutOfGas` network error.

An invalid or absent transaction from the blockchain is not picked up by the explorer, so Operations cannot be inflated with such transactions. I'll add a throw branch with a comment for these cases

Reply:
I think it's an additional reason to check the zero value : it's both functional and technical potential bug.